### PR TITLE
G-32 | Adds consent-request/on-init API & tests for it.

### PR DESCRIPTION
**Pull request by MounicaSagi**
* Refactors the response validator to accept CM_ID/HIU_ID and validate the response.

### CODE REVIEW

File: src/main/java/in/projecteka/gateway/clients/ConsentRequestServiceClient.java [hunk 1]
```diff
@@ -70,7 +70,20 @@
...
+                                webClientBuilder.build()
+                                        .post()
+                                        .uri(url + "/v1/consent-requests/on-init")
+                                        .contentType(MediaType.APPLICATION_JSON)
+                                        .header(HttpHeaders.AUTHORIZATION, token)
+                                        .bodyValue(serializedRequest)
+                                        .retrieve()
+                                        .onStatus(httpStatus -> !httpStatus.is2xxSuccessful(),
+                                                clientResponse -> Mono.error(ClientError.unableToConnect()))//TODO Error handling
+                                        .bodyToMono(Void.class)
```
Comment (ganesanarun) [inline src/main/java/in/projecteka/gateway/clients/ConsentRequestServiceClient.java]: you can use .toBodilessEntity to be more explicit.

File: src/main/java/in/projecteka/gateway/common/DefaultValidatedResponseAction.java [hunk 2]
```diff
@@ -11,26 +13,31 @@
...
     @Override
-    public Mono<Void> routeResponse(String xCmId, JsonNode updatedRequest) {
-        Optional<YamlRegistryMapping> configFor = cmRegistry.getConfigFor(xCmId);
+    public Mono<Void> routeResponse(String id, JsonNode updatedRequest) {
+        Optional<YamlRegistryMapping> configFor = id.equals(X_HIU_ID)
+                ? bridgeRegistry.getConfigFor(id, ServiceType.HIU)
+                : cmRegistry.getConfigFor(id);
         if (configFor.isEmpty()) {
-            logger.error("No mapping found for {}",xCmId);
+            logger.error("No mapping found for {}",id);
```
Comment (ganesanarun) [inline src/main/java/in/projecteka/gateway/common/DefaultValidatedResponseAction.java]:
```suggestion
            logger.error("No mapping found for {}", id);
```

File: src/main/java/in/projecteka/gateway/common/DefaultValidatedResponseAction.java [hunk 3]
```diff
@@ -11,26 +13,31 @@
...
         return serviceClient.routeResponse(updatedRequest,configFor.get().getHost());
     }
 
     @Override
-    public Mono<Void> handleError(Throwable throwable, String xCmId, JsonNode jsonNode) {
+    public Mono<Void> handleError(Throwable throwable, String id, JsonNode jsonNode) {
         //Does it make sense to call the same API back to notify only Error?
-        logger.error("Error in notifying CM with result",throwable);
+        logger.error("Error in notifying host with result",throwable);
         return Mono.empty();
```
Comment (ganesanarun) [inline src/main/java/in/projecteka/gateway/common/DefaultValidatedResponseAction.java:41]: jsonNode is not at all used. I see that it's coming from base class, if it's not used anywhere, just remove it please.

File: src/main/java/in/projecteka/gateway/common/ResponseOrchestrator.java [hunk 4]
```diff
@@ -10,16 +10,15 @@
...
     private static final Logger logger = LoggerFactory.getLogger(ResponseOrchestrator.class);
-    public Mono<Void> processResponse(HttpEntity<String> requestEntity) {
-        return validator.validateResponse(requestEntity)
+    public Mono<Void> processResponse(HttpEntity<String> requestEntity, String id) {
+        return validator.validateResponse(requestEntity, id)
                 .flatMap(validRequest -> {
                     JsonNode updatedJsonNode = Utils.updateRequestId(validRequest.getDeserializedJsonNode(),
                             validRequest.getCallerRequestId());
-                    return validatedResponseAction.execute(validRequest.getXCmId(),updatedJsonNode);
+                    return validatedResponseAction.execute(validRequest.getId(),updatedJsonNode);
```
Comment (ganesanarun) [inline src/main/java/in/projecteka/gateway/common/ResponseOrchestrator.java]:
```suggestion
                    return validatedResponseAction.execute(validRequest.getId(), updatedJsonNode);
```

File: src/main/java/in/projecteka/gateway/consent/ConsentRequestController.java [hunk 5]
```diff
@@ -28,6 +31,8 @@
     @ResponseStatus(HttpStatus.ACCEPTED)
     @PostMapping("/v1/consent-requests/on-init")
     public Mono<Void> onDiscoverCareContext(HttpEntity<String> requestEntity) {
+        Mono<Void> toBeFiredAndForgotten = consentResponseOrchestrator.processResponse(requestEntity, X_HIU_ID);
+        toBeFiredAndForgotten.subscribe();
```
Comment (ganesanarun) [inline src/main/java/in/projecteka/gateway/consent/ConsentRequestController.java]: If this is the pattern everywhere, why can't do the subscribe() inside the call itself?

File: src/main/java/in/projecteka/gateway/common/DefaultValidatedResponseAction.java (hunk 3 above)
Comment (MounicaSagi) [reply src/main/java/in/projecteka/gateway/common/DefaultValidatedResponseAction.java:41]: It's been used at another place to push the jsonNode as a message to an exchange.

File: src/main/java/in/projecteka/gateway/consent/ConsentRequestController.java (hunk 5 above)
Comment (MounicaSagi) [reply src/main/java/in/projecteka/gateway/consent/ConsentRequestController.java]: Made changes for callback API's. As this involves fixing tests, & I'm not familiar with the other API's. I left them the way they are now.

File: src/main/java/in/projecteka/gateway/clients/ConsentRequestServiceClient.java (hunk 1 above)
Comment (MounicaSagi) [reply src/main/java/in/projecteka/gateway/clients/ConsentRequestServiceClient.java]: Done.

**Review by ganesanarun — APPROVED** [review]

**Pull request merged**